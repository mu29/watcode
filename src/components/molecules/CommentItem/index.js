import React from 'react'
import PropTypes from 'prop-types'
import styled from 'styled-components'
import { Text } from 'components'
import { palette } from 'services/style'

const Wrapper = styled.div`
  display: flex;
`

const Container = styled.div`
  padding: 0.5rem;
  border-radius: 0.75rem;
  background-color: ${palette('gray.30')};
`

const Name = styled(Text)`
  margin-right: 0.25rem;
  font-weight: 600;
  color: ${palette('gray.100')};
`

const CommentItem = ({
  comment: {
    author,
    content,
  },
  ...props
}) => (
  <Wrapper { ...props }>
    <Container>
      <Name fontSize={ 12 }>{ author }</Name>
      <Text fontSize={ 12 }>{ content }</Text>
    </Container>
  </Wrapper>
)

CommentItem.propTypes = {
  comment: PropTypes.shape({
    author: PropTypes.string,
    content: PropTypes.string,
  }).isRequired,
}

export default CommentItem
