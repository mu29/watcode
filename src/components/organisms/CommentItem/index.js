import React from 'react'
import PropTypes from 'prop-types'
import styled from 'styled-components'
import { Text } from 'components'
import { CommentTooltip } from 'containers'
import { palette } from 'services/style'

const Wrapper = styled.div`
  display: flex;
`

const Container = styled.div`
  padding: 0.5rem;
  border-radius: 0.75rem;
  background-color: ${palette('gray.30')};
`

const Name = styled(Text)`
  margin-right: 0.25rem;
  font-weight: 600;
  color: ${palette('gray.100')};
`

const SytledCommentTooltip = styled(CommentTooltip)`
  display: none;

  ${Wrapper}:hover & {
    display: block;
  }
`

const CommentItem = ({
  artworkId,
  comment: {
    id,
    author,
    content,
  },
  ...props
}) => (
  <Wrapper { ...props }>
    <Container>
      <Name fontSize={ 12 }>{ author }</Name>
      <Text fontSize={ 12 }>{ content }</Text>
    </Container>
    <SytledCommentTooltip id={ id } artworkId={ artworkId } />
  </Wrapper>
)

CommentItem.propTypes = {
  artworkId: PropTypes.number.isRequired,
  comment: PropTypes.shape({
    author: PropTypes.string,
    content: PropTypes.string,
  }).isRequired,
}

export default CommentItem
