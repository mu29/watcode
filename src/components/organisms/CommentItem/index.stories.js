import React from 'react'
import { storiesOf } from '@storybook/react'
import CommentItem from '.'

const comment = {
  author: '영화번역가 황석희',
  content: '달인급 캐릭터 분량 조절의 기술. 이 긴 영화에 지루함 한 톨 없다. 너는 어른이 다 됐더라, 피터',
}

storiesOf('Organisms/CommentItem', module)
  .add('기본', () => (
    <CommentItem comment={ comment } />
  ))
