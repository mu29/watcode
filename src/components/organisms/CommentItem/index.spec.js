import React from 'react'
import CommentItem from '.'

const comment = {
  content: '달인급 캐릭터 분량 조절의 기술. 이 긴 영화에 지루함 한 톨 없다. 너는 어른이 다 됐더라, 피터',
  userAttributes: {
    name: '영화번역가 황석희',
  },
}

const component = (props = {}) => withTheme(<CommentItem comment={ comment } { ...props } />)

describe('<CommentItem />', () => {
  it('정상적으로 렌더링된다', () => {
    const wrapper = render(component())
    expect(wrapper).toMatchSnapshot()
  })
})
