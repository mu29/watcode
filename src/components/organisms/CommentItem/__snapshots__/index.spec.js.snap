// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<CommentItem /> 정상적으로 렌더링된다 1`] = `
.c1 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
}

.c2 {
  padding: 0.5rem;
  border-radius: 0.75rem;
}

.c0:hover .eBJlmb {
  display: block;
}

<div
  class="c0 c1"
>
  <div
    class="c2"
  >
    <span
      componentname="Text"
    />
    <span
      componentname="Text"
    >
      달인급 캐릭터 분량 조절의 기술. 이 긴 영화에 지루함 한 톨 없다. 너는 어른이 다 됐더라, 피터
    </span>
  </div>
  <span
    componentname="CommentTooltip"
  />
</div>
`;
